### PR TITLE
[CPM-150] Refactor and Modularise Options Menu

### PR DESCRIPTION
**Pull request by vee-christella**
## Changes

Separated Options Menu into 2 classes:
1. A wrapper `MenuWrapper` that contains both the options menu component and the `ConfirmationDialog` to trigger
2. The options menu component itself, but simplified

## Checklist

- [ ] Have tests been written?
- [x] Is the UI consistent?

## Requests
- [ ] Can somebody check the styling/CSS?

Resolves [CPM-150]

[CPM-152]: https://momentum-agile.atlassian.net/browse/CPM-152

[CPM-150]: https://momentum-agile.atlassian.net/browse/CPM-150

### CODE REVIEW

File: client/src/components/MenuWrapper.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import { Flex } from "@chakra-ui/core";
+import { OptionsMenu, ConfirmationDialog } from ".";
+
+const MenuWrapper = ({ item, itemType, setOpenConfirmationDialog, openConfirmationDialog, confirm, hasEdit, onEdit }) => {
```
Comment (Dinith1) [inline client/src/components/MenuWrapper.js:5]: Just a thought:

Couldn't you use the `onEdit` prop to infer `isEdit`? (so you wouldn't need the `isEdit` prop).

So in `OptionsMenu.js` maybe you could use `onEdit && ...`.

**Review by Dinith1 — APPROVED** [review]
Very nice! :shipit: Thank you for doing this 🥺 

Just left one comment, but it's not major (idk if my option good practice)

**Comment by daltonlim** [comment]
Lgtm

**Review by daltonlim — APPROVED** [review]
Lgtm

File: client/src/components/MenuWrapper.js (hunk 1 above)
Comment (vee-christella) [reply client/src/components/MenuWrapper.js:5]: Good point! 😮

**Review by FocalChord — APPROVED** [review]
Looks good Vee!

Just a question should a ticket be made for this so we don't have [Tech Debt]?

**Comment by vee-christella** [comment]
> Just a question should a ticket be made for this so we don't have [Tech Debt]?

Sure I can make it! I was just worried about our Burndown hahaha

**Pull request merged**